docs/sources/flow/reference/components/loki.source.podlogs.md

```diff
@@ -0,0 +1,238 @@
+following meta labels are available for relabeling:
+
+* `__meta_kubernetes_namespace`: The namespace of the Pod.
+* `__meta_kubernetes_pod_name`: The name of the Pod.
+* `__meta_kubernetes_pod_ip`: The pod IP of the Pod.
+* `__meta_kubernetes_pod_label_<labelname>`: Each label from the Pod.
+* `__meta_kubernetes_pod_labelpresent_<labelname>`: `true` for each label from
+  the Pod.
+* `__meta_kubernetes_pod_annotation_<annotationname>`: Each annotation from the
+  Pod.
+* `__meta_kubernetes_pod_annotationpresent_<annotationname>`: `true` for each
+  annotation from the Pod.
+* `__meta_kubernetes_pod_container_init`: `true` if the container is an
+  `InitContainer`.
+* `__meta_kubernetes_pod_container_name`: Name of the container.
+* `__meta_kubernetes_pod_container_image`: The image the container is using.
+* `__meta_kubernetes_pod_ready`: Set to `true` or `false` for the Pod's ready
+  state.
+* `__meta_kubernetes_pod_phase`: Set to `Pending`, `Running`, `Succeeded`, `Failed` or
+  `Unknown` in the lifecycle.
+* `__meta_kubernetes_pod_node_name`: The name of the node the pod is scheduled
+  onto.
+* `__meta_kubernetes_pod_host_ip`: The current host IP of the pod object.
+* `__meta_kubernetes_pod_uid`: The UID of the Pod.
+* `__meta_kubernetes_pod_controller_kind`: Object kind of the Pod's controller.
+* `__meta_kubernetes_pod_controller_name`: Name of the Pod's controller.
+
+In addition to the meta labels, the following labels are exposed to tell
+`loki.source.podlogs` which container to tail:
+
+* `__pod_namespace__`: The namespace of the Pod.
+* `__pod_name__`: The name of the Pod.
+* `__pod_container_name__`: The container name within the Pod.
+* `__pod_uid__`: The UID of the Pod.
+
+[LabelSelector]: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.24/#labelselector-v1-meta
+[RelabelConfig]: https://prometheus-operator.dev/docs/operator/api/#monitoring.coreos.com/v1.RelabelConfig
+
+## Blocks
+
+The following blocks are supported inside the definition of
+`loki.source.podlogs`:
+
+Hierarchy | Block | Description | Required
+--------- | ----- | ----------- | --------
+client | [client][] | Configures Kubernetes client used to tail logs. | no
+client > http_client_config | [http_client_config][] | HTTP client configuration for Kubernetes requests. | no
+selector | [selector][] | Label selector for which `PodLogs` to discover. | no
+selector > match_expression | [match_expression][] | Label selector expression for which `PodLogs` to discover. | no
+namespace_selector | [selector][] | Label selector for which namespaces to discover `PodLogs` in. | no
+namespace_selector > match_expression | [match_expression][] | Label selector expression for which namespaces to discover `PodLogs` in. | no
+
+The `>` symbol indicates deeper levels of nesting. For example, `client >
+http_client_config` refers to an `http_client_config` block defined
+inside a `client` block.
+
```

Comment:
I'm mentioning that users can supply multiple `match_expression` blocks, acting as AND clauses.

Reply:
Can you add that in the section for match_expression?